src/components/login/login.js

```diff
@@ -85,17 +86,33 @@ class Login extends React.Component {
     return data;
   }
 
-  // eslint-disable-next-line class-methods-use-this
   validatePassphrase(value) {
     const data = { passphrase: value };
-    if (!value || value === '') {
-      data.passphraseValidity = 'Empty passphrase';
-    } else {
-      data.passphraseValidity = isValidPassphrase(value) ? '' : 'Invalid passphrase';
-    }
+    data.passphraseValidity = !isValidPassphrase(value) ? this.getPassphraseValidationError(value) : '';
     return data;
   }
 
+  // eslint-disable-next-line class-methods-use-this
+  getPassphraseValidationError(passphrase) {
+    if (!passphrase) {
+      return 'Empty passphrase';
+    }
+    const mnemonic = passphrase.trim().toLowerCase().split(' ');
+    if (mnemonic.length < 12) {
+      return `Passphrase should have 12 words, entered passphrase has ${mnemonic.length}`;
+    }
+
+    const invalidWord = mnemonic.find(word => !inDictionary(word));
+    if (invalidWord && (invalidWord.length >= 2 && invalidWord.length <= 8)) {
+      const validWord = findSimilarWord(invalidWord);
+      if (validWord) {
```

Comment:
``Word "${invalidWord}" is not on the passphrase Word List.`` error message would be helpful when no `validWord` is found.